01-JS-DOM-APIs/js/index.js

```diff
@@ -0,0 +1,388 @@
+ * click, but if not, it'll do nothing. I know it's not efficient.*/
+function tellMeAJoke(){
+    JSuccess = false; ///* Set the global variable JSuccess to false. Tt's used
+    // to determinate if the server has answered and avoid showing old "time out" errors.
+    // If JSuccess = true, the server has answered. If not, the time out error, will be call.
+    var callId = Clicks;
+    var req = new XMLHttpRequest(); //A new HttpRequest object
+    req.open ("GET", "http://api.icndb.com/jokes/random", true); //Configuration: GET - src - Async
+    setTimeout(function(){
+        if (Clicks == callId && JSuccess == false){ //Is it the last call and now answer yet?
+            changeColor("red");
+            changeAdvice("ERROR: Time OUT!!");
+        }
+    },3000);
+    req.addEventListener("load", function(){ // When the server answer --> show the joke!!.
+        if (Clicks == callId){ //Is it the last call?
+            console.log("HERE" + Clicks);
+            showJoke(req); // This function it's the one that show JOKES. It's used by exercises 6 and 7.
+            changeColor("green");
+            changeAdvice("JOKE SUCCESS!!");
+            JSuccess = true;
+        }
+    });
+    req.send(null); // send the HttpRequest.
+}
+
+/* EX 7 (KEY) and 6
+ * function showJoke(reqJSON)
+ * where reqJSON -> It's a JSON object
+ *  This function takes a JSON answer of the HttpRequest, and show it in the HTML Document, modifying the DOM.
+*/
+function showJoke (reqJSON){
+    if (reqJSON.status < 400){ // if < 400 ->> Success!
+        var reqParse = JSON.parse(reqJSON.responseText); // parse the Json answer
+             /*Different commands i tried to understand the structure of the requested JSON object -
+             * i don't delete it in case somebody seeing the code wanna to know its structure.
+                 console.log(reqJSON.status, reqJSON.statusText); // Req status
+                 console.log(reqJSON.responseText); // Devolution
+                 console.log(reqParse); // [type (string), value (object)]
+                 console.log(reqParse.value); // [id (integer), joke(string), categories(array)]
+                 console.log(reqParse.value.categories); // categories elements
+             */
+        var elem = document.createElement("p"); // new element <p>
+        var cont = document.createTextNode(reqParse.value["joke"]); // new element <p> content
+        elem.appendChild(cont); // insert node cont in node elem <p>
+        var where = document.getElementById("hide"); // looking for the section where i will work
+        removeAllChildNodes(where); // this function remove all childs of a node, in this case, it will delete all the content
+        where.appendChild(elem.appendChild(cont)); // insert the <p> element with its content in the section "hide"
+        where.style.fontSize= "20px"; // Some style for a better watching
```

Comment:
This is fine but I'd suggest for next time to use a class instead and set the styling to that class.